src/kvstore/CompactionFilter.h

```diff
@@ -0,0 +1,98 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#ifndef KVSTORE_COMPACTIONFILTER_H_
+#define KVSTORE_COMPACTIONFILTER_H_
+
+#include "base/Base.h"
+#include <rocksdb/compaction_filter.h>
+#include "base/NebulaKeyUtils.h"
+#include "time/TimeUtils.h"
+#include "kvstore/Common.h"
+
+namespace nebula {
+namespace kvstore {
+
+class KVCompactionFilter final : public rocksdb::CompactionFilter {
+public:
+    KVCompactionFilter(GraphSpaceID spaceId, std::unique_ptr<KVFilter> kvFilter)
+        : spaceId_(spaceId)
+        , kvFilter_(std::move(kvFilter)) {
+    }
+
+    bool Filter(int,
+                const rocksdb::Slice& key,
+                const rocksdb::Slice& val,
+                std::string*,
+                bool*) const override {
+        return kvFilter_->filter(spaceId_,
```

Comment:
Could remove some of the parameters that are not used?

Reply:
It is rocksdb's interface.